### PR TITLE
scarthgap: add support for fTPM running as a trusted application in OP-TEE

### PR DESCRIPTION
**Pull request by sergioprado**
By inheriting the tdx-optee class and defining TDX_OPTEE_FTPM="1", users can enable support for a firmware TPM running as a TA (trusted application) in OP-TEE.

Build and runtime tested on Verdin iMX8MP.

### CODE REVIEW

**Review by rborn-tx — APPROVED** [review]
Nice addition!

**Comment by sergioprado** [comment]
Fixed the title of the PR, it was written kirkstone, but this is for scarthgap.

I also tested on kirkstone and it works as well. I will cherry-pick this commit after it is merged on scarthgap.

**Comment by rborn-tx** [comment]
@sergioprado I'll wait until the end of the day to give Jeremias a chance to review it.
@lucas-akira Pinging you just to keep you informed.

**Pull request merged**
